### PR TITLE
Auto dismiss rules swagger

### PR DESCRIPTION
**Pull request by yibirnba**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
```
Comment (nitsi) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json]: Test

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+  "info": {
+    "title": "Security Center",
+    "description": "API spec for Microsoft.Security (Azure Security Center) resource provider",
+    "version": "2019-01-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
```
Comment (nitsi) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json]: Test

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+        "state": {
+          "type": "string",
+          "description": "Possible states of the rule",
+          "x-ms-enum": {
+            "name": "State"
+          },
+          "enum": [
+            "Enabled",
+            "Disabled",
+            "Expired"
```
Comment (nitsi) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json]: What happens if we return expired but we don't set it here?

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+        },
+        "autoDismissScope": {
+          "type": "object",
+          "description": "The auto dismiss conditions",
+          "$ref": "#/definitions/AutoDismissScope"
+        }
+      },
+      "required": [
+        "alertType",
+        "expirationDate",
```
Comment (nitsi) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json]: Is it required? we can set a 6 month default

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+          "format": "date-time",
+          "description": "The last time this rule was modified"
+        },
+        "expirationDate": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Expiration date of the rule"
+        },
+        "reason": {
+          "type": "string",
```
Comment (nitsi) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json]: Should this need to be enum?

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissRules/GetAutoDismissRule_example.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      "body": {
+        "id": "/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/providers/Microsoft.Security/autoDismissRules/dismissIpAnomalyAlerts",
+        "name": "dismissIpAnomalyAlerts",
+        "type": "Microsoft.Security/autoDismissRules",
+        "properties": {
+          "alertType": "IpAnomaly",
+          "lastModifiedUTC": "2019-07-31T19:50:47.083633Z",
+          "expirationDateUTC": "2019-12-01T19:50:47.083633Z",
+          "state": "Enabled",
+          "reason": "Detects a normal activity (False Positive)",
```
Comment (nitsi) [inline specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissRules/GetAutoDismissRule_example.json]: Do we want to convert Reason into some kind of Enum? so the user will have easier time to send it?

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json (hunk 4 above)
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json]: This is up to us. WDYT?

Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json]: No , will be optional

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json (hunk 5 above)
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json]: The reasons probably will changed from time to time so we validate this in RP

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissRules/GetAutoDismissRule_example.json (hunk 6 above)
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/AutoDismissRules/GetAutoDismissRule_example.json]: The reasons probably will changed from time to time so we validate this in RP

File: specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json (hunk 3 above)
Comment (yibirnba) [reply specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/autoDismissRules.json]: will check with ARM

**Pull request merged**